src/components/NavBar/NavBar.tsx

```diff
@@ -203,10 +203,13 @@ export class NavBar extends OGSComponent<{}, any> {
     }
     updateOmnisearch = (ev) => {
         try {
-            let q = (ev.target.value && ev.target.value.trim()) || "";
+            let q = ev.target.value || "";
 
-            if (q !== this.state.omnisearch_string) {
+            if (q.trim() !== this.state.omnisearch_string.trim()) {
                 this.abortOmnisearch();
+            } else {
+                this.setState({ omnisearch_string: q });
```

Comment:
I don't think we need to set this setState here as we set it down elow in both the `q === ""` and `q !== ""` conditions